cdap-authorization/src/main/java/co/cask/common/authorization/client/SimpleAuthorizationClient.java

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.common.authorization.client;
+
+import co.cask.common.authorization.ACLEntry;
+import co.cask.common.authorization.ACLStore;
+import co.cask.common.authorization.IdentifiableObject;
+import co.cask.common.authorization.ObjectId;
+import co.cask.common.authorization.Permission;
+import co.cask.common.authorization.SubjectId;
+import co.cask.common.authorization.UnauthorizedException;
+import com.google.common.base.Function;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Predicate;
+import com.google.common.base.Supplier;
+import com.google.common.base.Suppliers;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Sets;
+import com.google.inject.Inject;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import javax.annotation.Nullable;
+
+/**
+ * Simple implementation of {@link AuthorizationClient} which uses {@link ACLStore}.
+ */
+public class SimpleAuthorizationClient implements AuthorizationClient {
+
+  private final Supplier<ACLStore> aclStoreSupplier;
+
+  /**
+   * @param aclStoreSupplier used to set and get {@link co.cask.common.authorization.ACLEntry}s
+   */
+  @Inject
+  public SimpleAuthorizationClient(ACLStoreSupplier aclStoreSupplier) {
+    this.aclStoreSupplier = aclStoreSupplier;
+  }
+
+  public SimpleAuthorizationClient(ACLStore aclStore) {
+    this.aclStoreSupplier = Suppliers.ofInstance(aclStore);
+  }
+
+  @Override
+  public void authorize(Iterable<ObjectId> objects, Iterable<SubjectId> subjects,
```

Comment:
This PR is based on the authorization core PR: https://github.com/caskdata/cdap/pull/1309